logic/src/commonMain/kotlin/com/wire/kalium/logic/functional/Either.kt

```diff
@@ -59,6 +59,17 @@ sealed class Either<out L, out R> {
             is Left -> fnL(value)
             is Right -> fnR(value)
         }
+
+    /**
+     * Applies fnL if this is a Left or fnR if this is a Right.
+     * @see Left
+     * @see Right
+     */
+    fun <T> safeFold(fnL: (L) -> T, fnR: (R) -> T): T =
+        when (this) {
+            is Left -> fnL(value)
+            is Right -> fnR(value)
+        }
```

Comment:
Just a question, I didn't try it. Could we achieve the same using the `fold` function from above?, and just forcing the return to value to be non-nullable, ie:

```suggestion
        fold(fnL, fnR)!!
```

Reply:
Nice suggestion @yamilmedina!

Food for thought: 
Personally I am not a fan of the current `fold` function and I would try to rename `safeFold` to `fold` and remove the current implementation.

I think that nowhere `fold` isn't being used with nullable values at the moment.

If the most common one is called `safeFold`, then we should make it the standard. `nullableFold` should be the alternative, not the default.

Reply:
i did change it to this
```kotlin 
    fun <T> fold(fnL: (L) -> T, fnR: (R) -> T): T = performFold(fnL, fnR)!!

    fun <T> nullableFold(fnL: (L) -> T?, fnR: (R) -> T?): T? = performFold(fnL, fnR)

    private inline fun <T> performFold(fnL: (L) -> T?, fnR: (R) -> T?): T? =
        when (this) {
            is Left -> fnL(value)
            is Right -> fnR(value)
        }
```